readers/combined.cc

```diff
@@ -689,42 +691,47 @@ mutation_reader make_combined_reader(schema_ptr schema,
         reader_permit permit,
         std::unique_ptr<reader_selector> selector,
         streamed_mutation::forwarding fwd_sm,
-        mutation_reader::forwarding fwd_mr) {
+        mutation_reader::forwarding fwd_mr,
+        combined_reader_statistics* statistics) {
     return make_mutation_reader<merging_reader<mutation_reader_merger>>(schema,
             std::move(permit),
             fwd_sm,
-            mutation_reader_merger(schema, std::move(selector), fwd_sm, fwd_mr));
+            mutation_reader_merger(schema, std::move(selector), fwd_sm, fwd_mr),
+            statistics);
 }
 
 mutation_reader make_combined_reader(schema_ptr schema,
         reader_permit permit,
         std::vector<mutation_reader> readers,
         streamed_mutation::forwarding fwd_sm,
-        mutation_reader::forwarding fwd_mr) {
+        mutation_reader::forwarding fwd_mr,
+        combined_reader_statistics* statistics) {
     if (readers.empty()) {
         return make_empty_flat_reader_v2(std::move(schema), std::move(permit));
     }
-    if (readers.size() == 1) {
+    if (readers.size() == 1 && !statistics) {
         return std::move(readers.front());
```

Comment:
should we rule out this optimization based on the presence of stats? I think we shouldn't.

Reply:
The statistics are collected in the combined reader. I use `mutation_fragment_batch` to get the number of sstables the row is read from. Thus I skip the optimization. Otherwise, I would need to propagate the statistics structure to a `mutation_reader` and I do not think it is necessary.

Note that the optimization is skipped only for compaction when statistics are required. For the rest of the reads, the optimization is not omitted.